torch/_inductor/codegen/common.py

```diff
@@ -209,9 +209,10 @@ def init_backend_registration():
 
     if get_scheduling_for_device("cuda") is None:
         # CUDACombinedScheduling combines Triton and CUDA C++ scheduling for CUDA devices via delegation
+        cuda_backends = {"triton": CUDACombinedScheduling, "halide": HalideScheduling}
         register_backend_for_device(
             "cuda",
-            CUDACombinedScheduling,
+            lambda *args, **kwargs: cuda_backends[config.cuda_backend](*args, **kwargs),
```

Comment:
Does simply `cuda_backends[config.cuda_backend]` without the lambda wrapper not work?

Reply:
It fails if you change the config value at runtime.